### ISSUE TITLE
Form submission (with relations) slows down after upgrade from 2.7.9 to 2.7.10

### ISSUE THREAD

**Issue by jakubbialas3a** [symfony/symfony#18011]
Form fields "entity" and "collection" are probably loading whole collection of target entities when form is submitted. In previous version (2.7.9) it was loading only one selected entity.

My form for adding new orders:

```php
    $builder
      ->add('name', 'text')
      ->add('client', 'entity', array('class' => 'CoreBundle:Client'))
```

And now I have over 15k clients, and they are all loaded as entities when I submit form, so it takes some unnecessary time and memory. Accually I get memory exhausted error, because it takes over 128MB, and in 2.7.9 it takes less than 15MB.

(I'm using form only for submitting rest api calls, not for rendering forms)

**Comment by backbone87** [comment]
Take a look at the ChoiceLoader interface, 
you can implement partial/lazy loading there

**Comment by webmozart** [comment]
This should be fixed by #17990. Could you please verify?

**Comment by weaverryan** [comment]
I just created a fresh 2.7.10 project with a Client entity and added 15000 rows. Initially, when submitting, it took 1200 ms every time. When I switched the code to the latest 2.7 branch - which included #17990 - it fixed the issue: my load time went down to about 100ms.

This issue can be closed.

**Comment by webmozart** [comment]
Thanks a lot @weaverryan :)
